app/scripts/modules/dcos/instance/details/details.controller.js

```diff
@@ -0,0 +1,216 @@
+          params.managedInstanceGroupName = instance.serverGroup;
+        }
+
+        params.namespace = instance.namespace;
+        instance.placement = {};
+
+        return instanceWriter.terminateInstance(instance, app, params);
+      };
+
+      confirmationModalService.confirm({
+        header: 'Really terminate ' + instance.instanceId + '?',
+        buttonText: 'Terminate ' + instance.instanceId,
+        account: instance.account,
+        provider: 'dcos',
+        taskMonitorConfig: taskMonitor,
+        submitMethod: submitMethod
+      });
+    };
+
+    this.registerInstanceWithLoadBalancer = function registerInstanceWithLoadBalancer() {
+      var instance = $scope.instance;
+      var loadBalancerNames = instance.loadBalancers.join(' and ');
+
+      var taskMonitor = {
+        application: app,
+        title: 'Registering ' + instance.taskId + ' with ' + loadBalancerNames
+      };
+
+      var submitMethod = function () {
+        return instanceWriter.registerInstanceWithLoadBalancer(instance, app, { interestingHealthProviderNames: ['Dcos'] } );
+      };
+
+      confirmationModalService.confirm({
+        header: 'Really register ' + instance.taskId + ' with ' + loadBalancerNames + '?',
+        buttonText: 'Register ' + instance.taskId,
+        account: instance.account,
+        taskMonitorConfig: taskMonitor,
+        submitMethod: submitMethod
+      });
+    };
+
+    this.deregisterInstanceFromLoadBalancer = function deregisterInstanceFromLoadBalancer() {
+      var instance = $scope.instance;
+      var loadBalancerNames = instance.loadBalancers.join(' and ');
+
+      var taskMonitor = {
+        application: app,
+        title: 'Deregistering ' + instance.taskId + ' from ' + loadBalancerNames
+      };
+
+      var submitMethod = function () {
+        return instanceWriter.deregisterInstanceFromLoadBalancer(instance, app, { interestingHealthProviderNames: ['Dcos'] } );
+      };
+
+      confirmationModalService.confirm({
+        header: 'Really deregister ' + instance.taskId + ' from ' + loadBalancerNames + '?',
+        buttonText: 'Deregister ' + instance.taskId,
+        provider: 'dcos',
+        account: instance.account,
+        taskMonitorConfig: taskMonitor,
+        submitMethod: submitMethod
+      });
+    };
+
+    this.canRegisterWithLoadBalancer = function() {
+      // var instance = $scope.instance;
+      // if (!instance.loadBalancers || !instance.loadBalancers.length) {
+      //   return false;
+      // }
+      // return instance.health.some(function(health) {
+      //   return health.state === 'OutOfService';
+      // });
+
+      return false;
+    };
+
+    this.canDeregisterFromLoadBalancer = function() {
+      return false;
+      // var instance = $scope.instance;
+      // if (!instance.loadBalancers || !instance.loadBalancers.length) {
+      //   return false;
+      // }
+      // return instance.healthState !== 'OutOfService';
+    };
```

Comment:
Is this TODO or does DCOS not support the notion of load balancer registration/deregistration at the instance level? If it's the latter, these methods should just be removed. If it's the former, a comment would be good here.